oligo_designer_toolsuite/oligo_specificity_filter/_filter_bowtie.py

```diff
@@ -181,17 +203,69 @@ def _find_hits(
         :param consider_hits_from_input_region: Flag to indicate whether hits from the input region should be considered.
         :type consider_hits_from_input_region: bool
         :return: A tuple containing a DataFrame of significant Bowtie hits and a list of oligos with these hits.
-        :rtype: (pd.DataFrame, list)
+        :rtype: pd.DataFrame
         """
         if consider_hits_from_input_region:
             # remove all hits where query and reference come from the same region
             search_results = search_results[
-                search_results["query_region_id"] != search_results["reference_region_id"]
+                search_results["query_region_id"]
+                != search_results["reference_region_id"]
             ]
 
-        oligos_with_hits = search_results["query"].unique()
+        return search_results
 
-        return search_results, oligos_with_hits
+    def _get_references(self, table_hits: pd.DataFrame, file_reference: str, region_id: str):
+        """
+        Retrieve the references sequences from the search results.
+
+        :param table_hits: DataFrame with the oligos that match the blast search.
+        :type table_hits: pd.DataFrame
+        :param file_reference: Path to the fasta file used as reference for the search.
+        :type file_reference: str
+        :param region_id: The identifier for the region within the database to filter.
+        :type region_id: str
+        :return: Reference sequences
+        :rtype: list
+        """
+
+        required_fields = [
+            "query",
+            "strand",
+            "reference",
+            "reference_start",
+            "query_sequence",
+        ]
+        if not all(filed in self.names_search_output for filed in required_fields):
+            raise ValueError(
+                f"Some of the required fields {required_fields} are missing in the search results."
+            )
+        table_hits["reference_end"] = table_hits.apply(lambda x: x["reference_start"] + len(x["query_sequence"]), axis=1)
+        table_hits["score"] = 0
+        bed = table_hits[["reference", "reference_start", "reference_end", "query", "score", "strand"]]
```

Comment:
consider creating a new data frame to avoid just having a view of table hits and altering it accidentially:
```
bed = pd.DataFrame({
    "chr": table_hits["reference"],
    "start": table_hits["reference_start"],
    "end": table_hits["reference_end"],
    "name": table_hits["query"],
    "score": 0,  
    "strand": table_hits["strand"],
})
```